tests/qxmpppushenableiq/tst_qxmpppushenableiq.cpp

```diff
@@ -0,0 +1,168 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include <QObject>
+#include "util.h"
+#include <QXmppPushEnableIq.h>
+#include <QXmppDataForm.h>
+
+class tst_QXmppPushEnableIq : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void testPushEnable();
+    void testPushDisable();
+    void testXmlNs();
+    void testDataForm();
+    void testIsEnableIq();
+};
+
+void tst_QXmppPushEnableIq::testPushEnable()
+{
+    const QByteArray xml(
+        R"(<iq id="x42" type="set">)"
+        R"(<enable xmlns="urn:xmpp:push:0" jid="push-5.client.example" node="yxs32uqsflafdk3iuqo"/>)"
+        "</iq>");
+
+    QXmppPushEnableIq iq;
+    parsePacket(iq, xml);
+    QCOMPARE(iq.mode(), QXmppPushEnableIq::Enable);
+    QCOMPARE(iq.jid(), "push-5.client.example");
+    QCOMPARE(iq.node(), "yxs32uqsflafdk3iuqo");
+
+    QXmppPushEnableIq sIq;
+    sIq.setJid("push-5.client.example");
+    sIq.setMode(QXmppPushEnableIq::Enable);
+    sIq.setNode("yxs32uqsflafdk3iuqo");
+    sIq.setType(QXmppIq::Set);
+    sIq.setId("x42");
+
+    serializePacket(sIq, xml);
+}
+
+void tst_QXmppPushEnableIq::testPushDisable()
+{
+    const QByteArray xml(
+        R"(<iq id="x97" type="set">)"
+        R"(<disable xmlns="urn:xmpp:push:0" jid="push-5.client.example" node="yxs32uqsflafdk3iuqo"/>)"
+        "</iq>");
+
+    QXmppPushEnableIq iq;
+    parsePacket(iq, xml);
+    QCOMPARE(iq.mode(), QXmppPushEnableIq::Disable);
+    QCOMPARE(iq.jid(), "push-5.client.example");
+
+    QXmppPushEnableIq sIq;
+    sIq.setJid("push-5.client.example");
+    sIq.setMode(QXmppPushEnableIq::Disable);
+    sIq.setNode("yxs32uqsflafdk3iuqo");
+    sIq.setType(QXmppIq::Set);
+    sIq.setId("x97");
+
+    serializePacket(sIq, xml);
+}
+
+void tst_QXmppPushEnableIq::testXmlNs()
+{
+    const QByteArray xml(
+        R"(<iq type="set" id="x97">)"
+        R"(<disable xmlns="urn:ympp:wrongns:0" jid="push-5.client.example"/>)"
+        "</iq>");
+
+    QXmppPushEnableIq iq;
+    parsePacket(iq, xml);
+    QVERIFY(iq.jid().isEmpty());
+}
+
+void tst_QXmppPushEnableIq::testDataForm()
+{
+    const QByteArray xml(
+        R"(<iq id="x43" type="set">)"
+        R"(<enable xmlns="urn:xmpp:push:0" jid="push-5.client.example" node="yxs32uqsflafdk3iuqo">)"
+        R"(<x xmlns="jabber:x:data" type="submit">)"
+        R"(<field type="hidden" var="FORM_TYPE"><value>http://jabber.org/protocol/pubsub#publish-options</value></field>)"
+        R"(<field type="text-single" var="secret"><value>eruio234vzxc2kla-91</value></field>)"
+        "</x>"
+        "</enable>"
+        "</iq>");
+
+    QXmppPushEnableIq iq;
+    parsePacket(iq, xml);
+    QVERIFY(!iq.dataForm().isNull());
+    QCOMPARE(iq.dataForm().fields().size(), 2);
+
```

Comment:
Please also serialize the `iq` to ensure parsing + serializing returns exactly the same XML again.